Makefile

```diff
@@ -0,0 +1,101 @@
+TOOLS_DIR="$(PWD)/.tools"
+
+GIT_COMMIT=$(shell git log -1 --format=%H)
+GIT_TAG=$(shell git symbolic-ref -q --short HEAD || git describe --tags --exact-match)
+BUILD_DATE=$(shell date -Ins)
+
+GOLANGCI_LINT_VERSION="v1.63.4"
+PROTOC_GEN_GO_VERSION="v1.36.2"
+PROTOC_GEN_GO_GRPC_VERSION="v1.5.1"
+PROTOC_GEN_GO_GRPC_GATEWAY_VERSION="v2.25.1"
+BUF_VERSION="v1.45.0"
+PROTOC_GEN_OPENAPIV2_VERSION="v2.15.2"
+
+.PHONY: all
+all: agent chorus proxy worker chorctl bench
+
+.PHONY: install-tools
+install-tools:
+	mkdir -p $(TOOLS_DIR)
+	GOBIN=$(TOOLS_DIR) go install golang.org/x/tools/cmd/goimports@latest
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(TOOLS_DIR) $(GOLANGCI_LINT_VERSION)
+	GOBIN=$(TOOLS_DIR) go install "google.golang.org/protobuf/cmd/protoc-gen-go@$(PROTOC_GEN_GO_VERSION)"
+	GOBIN=$(TOOLS_DIR) go install "google.golang.org/grpc/cmd/protoc-gen-go-grpc@$(PROTOC_GEN_GO_GRPC_VERSION)"
+	GOBIN=$(TOOLS_DIR) go install "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway@$(PROTOC_GEN_GO_GRPC_GATEWAY_VERSION)"
+	GOBIN=$(TOOLS_DIR) go install "github.com/bufbuild/buf/cmd/buf@$(BUF_VERSION)"
+	GOBIN=$(TOOLS_DIR) go install "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2@$(PROTOC_GEN_OPENAPIV2_VERSION)"
+
+.PHONY: ensure-tools
+ensure-tools:
+	@if [[ ! -f $(TOOLS_DIR)/buf || ! -f $(TOOLS_DIR)/protoc-gen-go || ! -f $(TOOLS_DIR)/protoc-gen-go-grpc || ! -f $(TOOLS_DIR)/protoc-gen-grpc-gateway || ! -f $(TOOLS_DIR)/protoc-gen-openapiv2 || ! -f $(TOOLS_DIR)/goimports || ! -f $(TOOLS_DIR)/golangci-lint ]]; then echo 'Some tools are mising. Execute "make install-tools".' && exit 1; fi;
+
+.PHONY: tidy
+tidy:
+	go mod tidy
+	cd tools/chorctl; go mod tidy
+	cd tools/bench; go mod tidy
+
+.PHONY: vet
+vet:
+	go vet ./...
+
+.PHONY: fmt
+fmt:
+	go fmt ./...
+
+.PHONY: imports
+imports:
```

Comment:
why do we need to call imports for every dir? what will happen if someone will add a new dir?

Reply:
It is made to exclude generated code, i.e. proto files, from formatting. 

`goimports` has an open issue to do this automatically, but so far it is not implemented.

Reply:
is there enough benefits of using`goimports ` in this case?
maybe we can introduce it later when it will be fixed?

Reply:
It unifies formatting of imports and simplifies navigation through import block.

I don't think that manual exclusion of a folder brings a lot of disadvantages.

Reply:
ok